pkgs/by-name/ls/lsp-ai/package.nix

```diff
@@ -0,0 +1,58 @@
+{
+  lib,
+  rustPlatform,
+  fetchFromGitHub,
+  pkg-config,
+  cmake,
+  openssl,
+  zlib,
+  perl,
+}:
+
+let
+  pname = "lsp-ai";
+  version = "0.7.0";
+
+  src = fetchFromGitHub {
+    owner = "SilasMarvin";
+    repo = pname;
+    rev = "66655f6285fe8aad8f2d72646e9ed47c06245525";
+    hash = "sha256-DwqqZBzLevuRCW6QzGyWdE+JtpW6b3EMDuiWtajv/U4=";
+  };
+
+in
+rustPlatform.buildRustPackage {
+  inherit pname version src;
+
+  cargoLock = {
+    lockFile = src + "/Cargo.lock";
+    allowBuiltinFetchGit = true;
+  };
+
+  nativeBuildInputs = [
+    pkg-config
+    cmake
+    perl
+  ];
+  buildInputs = [
+    openssl
+    openssl.dev
+    zlib
+  ];
+
+  buildFeatures = [ "all" ];
+
+  doCheck = false;
```

Comment:
If there's any individual test failure, try disable it: https://nixos.org/manual/nixpkgs/unstable/#disabling-package-tests

Reply:
individual test skips added.